src/OpenTracing/SpanContext.php

```diff
@@ -2,56 +2,18 @@
 
 namespace OpenTracing;
 
-use TracingContext\TracingContext;
-
-final class SpanContext
+use IteratorAggregate;
+
+/**
+ * SpanContext must be immutable in order to avoid complicated lifetime
+ * issues around Span finish and references.
+ *
+ * Baggage items are key:value string pairs that apply to the given Span,
+ * its SpanContext, and all Spans which directly or transitively reference
+ * the local Span. That is, baggage items propagate in-band along with the
+ * trace itself.
+ */
+interface SpanContext extends IteratorAggregate
 {
-    /**
-     * @var array
-     */
-    private $baggageItems;
-
-    /**
-     * @var Context
-     */
-    private $context;
-
-    private function __construct(Context $context)
-    {
-        $this->context = $context;
-    }
-
-    public static function create(Context $context)
-    {
-        return new self($context);
-    }
-
-    /**
-     * Creates a default SpanContext instance.
-     *
-     * @return SpanContext
-     */
-    public static function createAsDefault()
-    {
-        return new self(Context::createAsDefault());
-    }
-
-    /**
-     * Iterates over the baggage items.
-     *
-     * @param callable $closure
-     * @return array
-     */
-    public function foreachBaggageItem(callable $closure)
-    {
-        return array_map($closure, $this->baggageItems);
-    }
-
-    /**
-     * @return TracingContext
-     */
-    public function getContext()
-    {
-        return $this->context->getContext();
-    }
+    public function getBaggageItem($key);
```

Comment:
So apparently, `Span` should allow you to use `setBaggageItem` in which case, the `SpanContext` (which is immutable according to the spec) should be replaced with a new one including new baggage items. Does it makes sense to you guys? @felixfbecker @beberlei